projects/pre-commit.com/package.yml

```diff
@@ -0,0 +1,47 @@
+distributable:
+  url: https://github.com/pre-commit/pre-commit/archive/refs/tags/v{{ version }}.tar.gz
+  strip-components: 1
+
+# if there’s a github then we can parse the versions
+versions:
+  github: pre-commit/pre-commit/releases/tags  # reads github *releases* but uses the tags of those releases
+  strip: /^v/
+
+dependencies:
+  python.org: '>=3.8'
+
+build:
+  dependencies:
+    pip.pypa.io: '*'
+    gnu.org/bash: '>=2'  # for our venv/relocation scripts: feel free to fix that
+  script:
+    python-venv.sh {{prefix}}/bin/pre-commit
+
+provides:
+  - bin/pre-commit
+
+test:
+  dependencies:
+    git-scm.org: '*'
+  env:
+    GIT_AUTHOR_NAME : "test user"
+    GIT_AUTHOR_EMAIL : "test@example.com"
+    GIT_COMMITTER_NAME : "test user"
+    GIT_COMMITTER_EMAIL : "test@example.com"
+  script: |
+    test "$(pre-commit --version)" = "pre-commit {{version}}"
+    git config --global init.defaultBranch main # get rid of the warning
+    git init
+    mv $FIXTURE .pre-commit-config.yaml
+    pre-commit install
+    echo 'hi\n' > f.txt
+    git add f.txt
+    git commit f.txt -m "test"
```

Comment:
great test 🥰

Reply:
Shamelessly stolen from brew formula :laughing: 

https://github.com/Homebrew/homebrew-core/blob/24b557524b975cd7c5f458c147a1abcc9f511acf/Formula/pre-commit.rb

Reply:
Credit goes to @pharaujo